charms/jupyter-ui/src/charm.py

```diff
@@ -54,92 +58,107 @@ def send_info(self, event):
+                                "environment": {
+                                    'USERID_HEADER': 'kubeflow-userid',
+                                    'USERID_PREFIX': '',
+                                    'UI': config['ui'],
+                                    'URL_PREFIX': config['url-prefix'],
+                                    'DEV_MODE': config['dev-mode'],
                                 },
-                            ],
-                        }
-                    ]
-                },
-                "containers": [
-                    {
-                        "name": "jupyter-ui",
-                        "imageDetails": image_details,
-                        'ports': [{'name': 'http', 'containerPort': config['port']}],
-                        "envConfig": {
-                            'USERID_HEADER': 'kubeflow-userid',
-                            'USERID_PREFIX': '',
-                            'UI': config['ui'],
-                            'URL_PREFIX': config['url-prefix'],
-                            'DEV_MODE': config['dev-mode'],
+                            }
                         },
-                        "volumeConfig": [
-                            {
-                                "name": "config",
-                                "mountPath": "/etc/config",
-                                "files": [
-                                    {
-                                        "path": "spawner_ui_config.yaml",
-                                        "content": Path('src/spawner_ui_config.yaml').read_text(),
-                                    }
-                                ],
-                            },
-                        ],
-                    }
-                ],
-            },
+                }
+        config_template = None
+        with open(Path('src/spawner_ui_config.yaml')) as file:
+            config_template = yaml.full_load(file)
+        # Configure jupyter notebook url list
+        if config['default_notebook_lists'] != "default":
+            config_template['spawnerFormDefaults']['image']['options'] = \
+                                            config['default_notebook_lists'].split(',')
+            config_template['spawnerFormDefaults']['image']['value'] = \
+                                            config['default_notebook_lists'].split(',')[0]
+        # Add a Pebble config layer to the scraper container
+        container = self.unit.get_container("jupyter-ui")
+        container.push("/etc/config/spawner_ui_config.yaml", yaml.dump(config_template))
+        container.add_layer("jupyter-ui", layer, combine=True)
+        # Check if the scraper service is already running and start it if not
+        if not container.get_service("jupyter-ui").is_running():
+            container.start("jupyter-ui")
+            logger.info("Jupyter-ui service started")
+
+    def _k8s_auth(self) -> bool:
```

Comment:
This could be a new functionality in the kubernetes-rapper library as again, this seems like something most of the charms will do.